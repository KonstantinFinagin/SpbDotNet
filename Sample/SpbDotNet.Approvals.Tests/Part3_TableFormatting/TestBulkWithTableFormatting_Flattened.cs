using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Extensions;
using SpbDotNet.Approvals.Tests.Helpers;

namespace SpbDotNet.Approvals.Tests.Part3_TableFormatting
{
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Results")]
    public class TestBulkWithTableFormatting_Flattened
    {
        [Fact]
        public void Approves_BulkOutput_AsAsciiTable_Flattened()
        {
            // Arrange
            var inputs = new[]
            {
                new InputData { UserId = 1, RawInput = "Racecar" },
                new InputData { UserId = 2, RawInput = "Test123" },
                new InputData { UserId = 3, RawInput = "Hello World!" }
            };

            var analyzer = AnalyzerMocks.GetAnalyzerMock();
            var processor = new Processor(analyzer);

            // Act
            var results = processor.ProcessBulk(inputs);

            // Assert

            var sb = new StringBuilder();

            sb.AppendLine("Main info:");
            var flattened = results.Select(Flatten).ToList();
            var formattedMain = AsciiTableFormatter.Format(flattened);
            sb.AppendSafeWithNewLine(formattedMain);

            sb.AppendLine("External info:");
            var externalInfo = results.Select(r => r.ExternalInfo);
            var formattedExternal = AsciiTableFormatter.Format(externalInfo);
            sb.AppendSafeWithNewLine(formattedExternal);

            ApprovalTests.Approvals.Verify(sb.ToString());
        }

        #region formatters

        private static object Flatten(OutputData data)
        {
            // Creating an anonymous object with simple key-value pairs
            return new
            {
                UserId = data.UserId,
                NormalizedInput = data.NormalizedInput,
                InputLength = data.InputLength,
                ContainsDigits = data.ContainsDigits,
                ReversedInput = data.ReversedInput,
                ProcessedAt = data.ProcessedAt,
                Hash = data.Hash,
                IsPalindrome = data.IsPalindrome,
                VowelCount = data.Metadata?.VowelCount,
                ConsonantCount = data.Metadata?.ConsonantCount,
                CharFrequency = data.Metadata?.CharFrequency,
                RawProcessingLog = data.RawProcessingLog
            };
        }

        #endregion formatters
    }
}
