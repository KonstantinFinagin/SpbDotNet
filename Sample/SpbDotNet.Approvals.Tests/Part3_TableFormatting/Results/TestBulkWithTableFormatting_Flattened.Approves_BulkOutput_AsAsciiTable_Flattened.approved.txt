Main info:
+--------+-----------------+-------------+----------------+---------------+------------------------+------------------+--------------+------------+----------------+------------------------------------------------------+------------------------------------+
| UserId | NormalizedInput | InputLength | ContainsDigits | ReversedInput | ProcessedAt            | Hash             | IsPalindrome | VowelCount | ConsonantCount | CharFrequency                                        | RawProcessingLog                   |
+--------+-----------------+-------------+----------------+---------------+------------------------+------------------+--------------+------------+----------------+------------------------------------------------------+------------------------------------+
| 1      | racecar         | 8           | False          | racecar       | 01/01/2024 12:00:00 am | cmFjZWNhcg==     | True         | 3          | 4              | r: 2, a: 2, c: 2, e: 1                               | Processing data for 1: lorem ipsum |
| 2      | test123         | 8           | True           | 321tset       | 01/01/2024 12:00:00 am | dGVzdDEyMw==     | False        | 1          | 3              | t: 2, e: 1, s: 1, 1: 1, 2: 1, 3: 1                   | Processing data for 2: lorem ipsum |
| 3      | hello world!    | 13          | False          | !dlrow olleh  | 01/01/2024 12:00:00 am | aGVsbG8gd29ybGQh | False        | 3          | 7              | h: 1, e: 1, l: 3, o: 2,  : 1, w: 1, r: 1, d: 1, !: 1 | Processing data for 3: lorem ipsum |
+--------+-----------------+-------------+----------------+---------------+------------------------+------------------+--------------+------------+----------------+------------------------------------------------------+------------------------------------+

External info:
+----------+-------+------------+
| Category | Score | Tags       |
+----------+-------+------------+
| mocked   | 0.9   | type: test |
| mocked   | 0.9   | type: test |
| mocked   | 0.9   | type: test |
+----------+-------+------------+

